File: src/main/java/unificado/core/service/cuentanegocio/model/CuentaNegocio.java
package unificado.core.service.cuentanegocio.model;

import jakarta.persistence.*;
import lombok.*;
import unificado.core.service.cuenta.model.Cuenta;
import unificado.core.service.model.Auditable;
import unificado.core.service.negocio.model.Negocio;

import java.math.BigDecimal;

@Data
@Entity
@Table(name = "cuentanegocio", uniqueConstraints = @UniqueConstraint(columnNames = {"cun_neg_id", "cun_cuenta"}))
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class CuentaNegocio extends Auditable {

    @Id
    @Column(name = "clave")
    private Long cuentaNegocioId;

    @Column(name = "cun_neg_id")
    private Integer negocioId;

    @Column(name = "cun_cuenta")
    private Long numeroCuenta;

    @Column(name = "cun_maestro")
    private BigDecimal numeroMaestro;

    @Column(name = "cun_nombre")
    private String nombre;

    @OneToOne
    @JoinColumn(name = "cun_neg_id", referencedColumnName = "neg_id", insertable = false, updatable = false)
    private Negocio negocio;

    @OneToOne
    @JoinColumn(name = "cun_maestro", referencedColumnName = "pla_cuenta", insertable = false, updatable = false)
    private Cuenta cuenta;

}

File: src/main/java/unificado/api/rest/kotlin/model/CuentaNegocio.kt
package unificado.api.rest.kotlin.model

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.OneToOne
import jakarta.persistence.Table
import jakarta.persistence.UniqueConstraint
import java.math.BigDecimal

@Entity
@Table(name = "cuentanegocio", uniqueConstraints = [UniqueConstraint(columnNames = ["negocioId", "numeroCuenta"])])
data class CuentaNegocio(

    @Id
    @Column(name = "clave")
    var cuentaNegocioId: Long? = null,

    @Column(name = "cun_neg_id")
    var negocioId: Int? = null,

    @Column(name = "cun_cuenta")
    var numeroCuenta: Long? = null,

    @Column(name = "cun_maestro")
    var numeroMaestro: BigDecimal? = null,

    @Column(name = "cun_nombre")
    var nombre: String = "",

    @OneToOne(optional = true)
    @JoinColumn(name = "cun_neg_id", insertable = false, updatable = false)
    var negocio: Negocio? = null,

    @OneToOne(optional = true)
    @JoinColumn(name = "cun_maestro", insertable = false, updatable = false)
    var cuenta: Cuenta? = null

) : Auditable()
